cl/lib/elasticsearch_utils.py

```diff
@@ -3243,14 +3246,6 @@ def do_es_sweep_alert_query(
 
     limit_inner_hits({}, main_results, cd["type"])
     set_results_highlights(main_results, cd["type"])
-
-    for result in main_results:
-        child_result_objects = []
-        if hasattr(result, "child_docs"):
-            for child_doc in result.child_docs:
-                child_result_objects.append(child_doc.to_dict())
-            result["child_docs"] = child_result_objects
```

Comment:
Good question! I removed this code because it was redundant. The `do_es_sweep_alert_query` method already calls `limit_inner_hits`, which sets the `child_docs` key in the results. Therefore, this code was no longer necessary. I realized that we only need to perform this conversion to use a `defaultdict` on `child_docs` when the results are an `AttrDict` instance, and they are going to be serialized for rendering an API response. That’s why it’s not required here.

This observation led me to realize that `set_child_docs_and_score` was more complex than it needed to be. If the results are already a dictionary, there’s no need to convert them again. As a result, I refactored that method and also moved the logic for assigning the `BM25` score here.